### PR TITLE
feat: document list

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

File: src/components/DocumentList/VerifiedLabel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { render } from "@testing-library/react-native";
+import { VerifiedLabel } from "./VerifiedLabel";
+
+jest.mock("@expo/vector-icons", () => ({ Feather: "Feather Icons" }));
+
+describe("VerifiedLabel", () => {
+  it("should show VERIFIED when isVerified is true", async () => {
+    expect.assertions(1);
```
Comment (sebastianquek) [inline src/components/DocumentList/VerifiedLabel.test.tsx]: Do we need these assertions given that the render isn't async?

File: src/components/DocumentRenderer/WebViewFrame.tsx [hunk 2]
```diff
@@ -10,22 +10,22 @@ export interface WebViewFrameRef {
...
 
-const WebViewFrame: FunctionComponent<WebViewFrameProps> = ({
+export const WebViewFrame: FunctionComponent<WebViewFrame> = ({
   document,
   onTemplateMessageHandler,
   setGoToTemplate
-}: WebViewFrameProps) => {
-  const webViewRef = useRef();
+}: WebViewFrame) => {
+  const webViewRef = useRef<WebView>() as RefObject<WebView>;
```
Comment (sebastianquek) [inline src/components/DocumentRenderer/WebViewFrame.tsx]: We can avoid casting:
```suggestion
  const webViewRef: RefObject<WebView> = useRef<WebView>(null);
```

File: src/components/ScreenView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactElement } from "react";
+import { SafeAreaView, StatusBar } from "react-native";
+
+interface ScreenView {
+  children: ReactElement;
+}
+
+export const ScreenView = ({ children }: ScreenView): ReactElement => (
```
Comment (sebastianquek) [inline src/components/ScreenView/index.tsx]: Don't need to explicitly define the interface if we turn off the prop-types linting. `export const ScreenView: FunctionComponent = ({ children }) => (`

File: src/components/DocumentList/DocumentList.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export interface DocumentList {
+  documents: DocumentItem[];
+  navigateToDoc: (documentId: string) => void;
+}
+
+export const DocumentList = ({
+  documents,
+  navigateToDoc
+}: DocumentList): ReactElement => {
```
Comment (sebastianquek) [inline src/components/DocumentList/DocumentList.tsx]: We can utilise `FunctionComponent<DocumentList>` then we don't need to explicitly define the return type.

File: src/components/DocumentList/DocumentList.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  const renderedDocumentListItem = documents.map(
+    (doc): ReactElement => (
+      <DocumentListItem
+        key={doc.id}
+        title={doc.title}
+        isVerified={doc.isVerified}
+        lastVerification={doc.lastVerification}
+        onPress={(): void => navigateToDoc(doc.id)}
```
Comment (sebastianquek) [inline src/components/DocumentList/DocumentList.tsx]: The return type of functions can also be removed here too. Don't want to be defining the return type of every anonymous function. Let's add 
```
"@typescript-eslint/explicit-function-return-type": [
        "warn",
        {
          "allowExpressions": true,
          "allowTypedFunctionExpressions": true
        }
      ],
```
to our eslint too. (https://github.com/Open-Attestation/react-component-template/blob/master/package.json#L130)

File: src/components/DocumentList/VerifiedLabel.test.tsx (hunk 1 above)
Comment (sebastianquek) [reply src/components/DocumentList/VerifiedLabel.test.tsx]: Nvm, looks like we're using the entire jest eslint plugin.

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

File: src/components/DocumentList/VerifiedLabel.test.tsx (hunk 1 above)
Comment (yehjxraymond) [reply src/components/DocumentList/VerifiedLabel.test.tsx]: yea...

File: src/components/DocumentRenderer/WebViewFrame.tsx (hunk 2 above)
Comment (yehjxraymond) [reply src/components/DocumentRenderer/WebViewFrame.tsx]: ah I've tried this but without the null so it was funky, with the null it fixed it. Thanks

File: src/components/ScreenView/index.tsx (hunk 3 above)
Comment (yehjxraymond) [reply src/components/ScreenView/index.tsx]: Ah great, this works

File: src/components/DocumentList/DocumentList.tsx (hunk 4 above)
Comment (yehjxraymond) [reply src/components/DocumentList/DocumentList.tsx]: image

Problem is that it is asking for props validation now that the return type has been removed.

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=10

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-10

**Pull request merged**